feature: translate tutorial part two. (#121)
gatsbyjs/gatsby-pt-BR: docs/tutorial/part-two/index.md

3. Inicie o servidor de desenvolvimento:

```shell
gatsby develop
```

Se você abrir seu projeto no seu projeto no navegador, deverá ver um plano de fundo lavanda aplicado no "hello world" starter:

![Lavender Hello World!](global-css.png)

> Dica: Esta parte do tutorial se concentrou na maneira mais rápida e direta de começar a estilizar um site do Gatsby - importando arquivos CSS padrão diretamente, usando o `gatsby-browser.js`. Na maioria dos casos, a melhor maneira de adicionar estilos globais é com um componente de layout compartilhado. [Confira a documentação](/docs/global-css/) para saber mais sobre essa abordagem.

## Usando css com escopo de componente

Até agora, falamos sobre a abordagem mais tradicional do uso padrão de CSS. Agora, falaremos sobre vários métodos de modularização do CSS para lidar com o estilo de maneira orientada a componentes

### CSS Modules

Vamos explorar o **CSS Modules**. Citando de
[the CSS Module homepage](https://github.com/css-modules/css-modules):

> A **CSS Module** é um arquivo CSS no qual todos os nomes de classe e de animação
> têm escopo localmente por padrão.

> CSS Modules são muito populares porque eles deixam você escrecer CSS normalmente, mas com muito mais egurança. A ferramenta gera classes únicas e nomes de animações de forma automática, então você não precisa se preocupar com conflitos entre os seletores.

Reviewer: ```suggestion
CSS Modules são muito populares porque eles deixam você escrecer CSS normalmente, mas com muito mais segurança. A ferramenta gera classes únicas e nomes de animações de forma automática, então você não precisa se preocupar com conflitos entre os seletores.
```

> Gatsby trabalha fora da caixa com CSS Modules, Esta abordagem é altamente recomendada para novos projetos com Gatbsy(e React no geral).

Reviewer: ```suggestion
Gatsby trabalha fora da caixa com CSS Modules, Esta abordagem é altamente recomendada para novos projetos com Gatbsy (e React no geral).
```

#### ✋ Criandou uma nova página usando CSS Modules

Nesta seção, você vai criar uma nova página e estiliza-lá usando CSS Module.

Primeiro, crie um novo componente chamado `Container`.

> 1. Cria uma nova pasta em `src/components` e então, in neste diretório, cria um arquivo chamado `container.js` e cole o seguinte código:

Reviewer: ```suggestion
1. Cria uma nova pasta em `src/components` e então, dentro deste diretório, crie um arquivo chamado `container.js` e cole o seguinte código:
```

```javascript:title=src/components/container.js

> import React from "react";

Reviewer: ```suggestion
import React from "react"
```

> import containerStyles from "./container.module.css";

Reviewer: ```suggestion
import containerStyles from "./container.module.css"
```

export default ({ children }) => (
  <div className={containerStyles.container}>{children}</div>

> );

Reviewer: ```suggestion
)
```

```

> Você notará que importou um arquivo de módulo css chamado`container.module.css`. Vamos criar esse arquivo agora.

Reviewer: ```suggestion
Você notará que importou um arquivo de módulo css chamado `container.module.css`. Vamos criar esse arquivo agora.
```

}
```

Você noratá que o arquivo tem `.module.css` no final do seu nome em vez do usual `.css`. Esta é a forma como avisamos ao Gatsby que este arquivo deve ser processado como um CSS Module em vez de CSS comum.

3. Crie uma nova página criando um arquivo em:
   `src/pages/about-css-modules.js`:

```javascript:title=src/pages/about-css-modules.js

> import React from "react";

Reviewer: ```suggestion
import React from "react"
```

> import Container from "../components/container";

Reviewer: ```suggestion
import Container from "../components/container"
```

export default () => (
  <Container>
    <h1>About CSS Modules</h1>
    <p>CSS Modules are cool</p>
  </Container>

> );

Reviewer: ```suggestion
)
```

```

> Agore, se você visitar, `http://localhost:8000/about-css-modules/`, sua página deve ficar da seguinte forma:

Reviewer: ```suggestion
Agore, se você visitar `http://localhost:8000/about-css-modules/`, sua página deve ficar da seguinte forma:
```

}
```

> Você noratá que o arquivo tem `.module.css` no final do seu nome em vez do usual `.css`. Esta é a forma como avisamos ao Gatsby que este arquivo deve ser processado como um CSS Module em vez de CSS comum.

Reviewer 2: poderia alterar o `noratá` ?

}
```

3. Importe o novo arquivo `src/pages/about-css-modules.module.css` na página `about-css-modules.js` que você criou mais cedo editando as primeiras linhas da seguinte forma:

```javascript:title=src/pages/about-css-modules.js

> import React from "react";

Reviewer: ```suggestion
import React from "react"
```

// highlight-next-line

> import styles from "./about-css-modules.module.css";

Reviewer: ```suggestion
import styles from "./about-css-modules.module.css"
```

> import Container from "../components/container";

Reviewer: ```suggestion
import Container from "../components/container"
```

// highlight-next-line

> console.log(styles);

Reviewer: ```suggestion
console.log(styles)
```

```

> O código `console.log (styles)` mostrará a importação resultante para que você possa ver o resultado do seu arquivo `. / About-css-modules.module.css` processado. Se você abrir o console do desenvolvedor (usando por exemplo, Firefox ou Chrome developer tools) em seu navefador, você vai ver:

Reviewer: ```suggestion
O código `console.log(styles)` mostrará a importação resultante para que você possa ver o resultado do seu arquivo `. /about-css-modules.module.css` processado. Se você abrir o console do desenvolvedor (usando por exemplo, Firefox ou Chrome developer tools) em seu navegador, você vai ver:
```

![Import result of CSS module in console](css-modules-console.png)

> Se você comprarar isso com seu arquivo CSS, você vai ver que cada classe agora é uma chave no objeto importado apontando para uma longa string, por exemplo, `avatar` aponta para `src-pages----about-css-modules-module---avatar---2lRF7`. Essas são os nomes das classes geradas pelo CSS Modules. Elas são únicas em todo seu site. E como você precisa importá-los para usar as classes,nunca há dúvidas sobre onde algum CSS está sendo usado.

Reviewer: ```suggestion
Se você comprarar isso com seu arquivo CSS, você vai ver que cada classe agora é uma chave no objeto importado apontando para uma longa string, por exemplo, `avatar` aponta para `src-pages----about-css-modules-module---avatar---2lRF7`. Essas são os nomes das classes geradas pelo CSS Modules. Elas são únicas em todo seu site. E como você precisa importá-los para usar as classes, nunca há dúvidas sobre onde algum CSS está sendo usado.
```

4. Crie um componente `User`.

Crie um novo componente `<User />` inline na página `about-css-modules.js`. Modifique `about-css-modules.js` para ficar parecido com o seguinte código:

```jsx:title=src/pages/about-css-modules.js

> import React from "react";

Reviewer: ```suggestion
import React from "react"
```

> import styles from "./about-css-modules.module.css";

Reviewer: ```suggestion
import styles from "./about-css-modules.module.css"
```

> import Container from "../components/container";

Reviewer: ```suggestion
import Container from "../components/container"
```

> console.log(styles);

Reviewer: ```suggestion
console.log(styles)
```

console.log(styles)
```

O código `console.log(styles)` mostrará a importação resultante para que você possa ver o resultado do seu arquivo `. /about-css-modules.module.css` processado. Se você abrir o console do desenvolvedor (usando por exemplo, Firefox ou Chrome developer tools) em seu navegador, você vai ver:

![Import result of CSS module in console](css-modules-console.png)

> Se você comprarar isso com seu arquivo CSS, você vai ver que cada classe agora é uma chave no objeto importado apontando para uma longa string, por exemplo, `avatar` aponta para `src-pages----about-css-modules-module---avatar---2lRF7`. Essas são os nomes das classes geradas pelo CSS Modules. Elas são únicas em todo seu site. E como você precisa importá-los para usar as classes, nunca há dúvidas sobre onde algum CSS está sendo usado.

Reviewer 2: poderia arrumar esta paravra `comprarar` ?

> Se você comprarar isso com seu arquivo CSS, você vai ver que cada classe agora é uma chave no objeto importado apontando para uma longa string, por exemplo, `avatar` aponta para `src-pages----about-css-modules-module---avatar---2lRF7`. Essas são os nomes das classes geradas pelo CSS Modules. Elas são únicas em todo seu site. E como você precisa importá-los para usar as classes, nunca há dúvidas sobre onde algum CSS está sendo usado.

Author: ```suggestion
Se você comparar isso com seu arquivo CSS, você vai ver que cada classe agora é uma chave no objeto importado apontando para uma longa string, por exemplo, `avatar` aponta para `src-pages----about-css-modules-module---avatar---2lRF7`. Essas são os nomes das classes geradas pelo CSS Modules. Elas são únicas em todo seu site. E como você precisa importá-los para usar as classes, nunca há dúvidas sobre onde algum CSS está sendo usado.
```

      <p className={styles.excerpt}>{props.excerpt}</p>
    </div>
  </div>

> );

Reviewer: ```suggestion
)
```

    />
    {/* highlight-end */}
  </Container>

> );

Reviewer: ```suggestion
)
```

Agora a página final deve ficar assim:

![User list page with CSS modules](css-modules-userlist.png)

### CSS-in-JS

CSS-in-JS é uma abordagem de estilo orientada a componentes. Geralmente, é um padrão em que [CSS é composto inline usando JavaScript](https://reactjs.org/docs/faq-styling.html#what-is-css-in-js).

#### Usando CSS-in-JS com Gatsby

Existem muitas bibliotecas CSS-in-JS diferentes e muitas delas já possuem plugins Gatsby. Não abordaremos um exemplo de CSS-in-JS neste tutorial inicial, mas recomendamos que você [explore](/docs/styling/) o que o ecossistema tem a oferecer. Existem minitutoriais para duas bibliotecas, em particular, [Emotion](/docs/emotion/) e [Styled Components](/docs/styled-components/).

#### Sugestão de leitura em CSS-in-JS

Se você estiver interessado em ler mais, confira[Christopher "vjeux" Chedeau's 2014 presentation that sparked this movement](https://speakerdeck.com/vjeux/react-css-in-js) bem como [Mark Dalgleish's more recent post "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).

### Outras opções de CSS

> O Gatsby suporta quase todas as opções de estilo possíveis (se ainda não houver um plug-in para a sua opção CSS favorita, [por favor contribua com um!](/contributing/how-to-contribute/))

Reviewer: ```suggestion
O Gatsby suporta quase todas as opções de estilo possíveis (se ainda não houver um plugin para a sua opção CSS favorita, [por favor contribua com um!](/contributing/how-to-contribute/))
```
